src/test/java/xades4j/production/SignerSpecificTest.java

```diff
@@ -0,0 +1,130 @@
+package xades4j.production;
+
+import org.bouncycastle.asn1.ASN1Encodable;
+import org.bouncycastle.asn1.DERBMPString;
+import org.bouncycastle.asn1.DERUTF8String;
+
+
+import org.bouncycastle.asn1.x500.AttributeTypeAndValue;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x500.X500NameBuilder;
+
+import org.bouncycastle.asn1.x500.style.RFC4519Style;
+
+import org.bouncycastle.asn1.x509.BasicConstraints;
+import org.bouncycastle.asn1.x509.Extension;
+import org.bouncycastle.asn1.x509.KeyUsage;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import xades4j.algorithms.EnvelopedSignatureTransform;
+import xades4j.properties.DataObjectDesc;
+
+import xades4j.providers.impl.DirectKeyingDataProvider;
+
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.math.BigInteger;
+
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.SecureRandom;
+import java.security.Security;
+import java.security.cert.X509Certificate;
+import java.util.Date;
+import java.util.Random;
+import java.util.ResourceBundle;
+
+
+/**
+ * @author Artem R. Romanenko
+ * @version 02/04/2018
+ */
+public class SignerSpecificTest extends SignerBESTest {
+    private final static ResourceBundle BUNGLE = ResourceBundle.getBundle(SignerSpecificTest.class.getName());
+
+    @Test
+    public void signWithNationalCertificate() throws Exception {
+        Security.addProvider(new BouncyCastleProvider());
+        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA", BouncyCastleProvider.PROVIDER_NAME);
+        keyGen.initialize(1024, new SecureRandom());
+        Date validityBeginDate = new Date(System.currentTimeMillis() - 24 * 60 * 60 * 1000);
+        long add = (1L * 365L * 24L * 60L * 60L * 1000L);  //1 year
+        Date validityEndDate = new Date(System.currentTimeMillis() + add);
+
+
+        String nationalOrganizationName = "National organization name '" + BUNGLE.getString("common.name") + "'";
+        DERBMPString derbmpString = new DERBMPString(nationalOrganizationName);
+        DERUTF8String derutf8String = new DERUTF8String(nationalOrganizationName);
+        ASN1Encodable[] strings = new ASN1Encodable[]{derbmpString, derutf8String};
```

Comment:
Could each item of this list be a parameter of the test, using JUnit's [parameterized tests](https://github.com/junit-team/junit4/wiki/parameterized-tests)? Makes it easier to add new test subjects.

Reply:
I fixed it.